ts/graphs/preferences.ts

```diff
@@ -58,7 +58,11 @@ function preparePreferences(
         setGraphPreferences(constructPreferences());
     }
 
-    for (const [key, value] of Object.entries(graphsPreferences)) {
+    for (const [key, value] of Object.entries(
+        pb.BackendProto.GraphsPreferences.toObject(graphsPreferences, {
+            defaults: true,
```

Comment:
I have to say this confuses me quite a bit. I thought the defaults logic was implemented in Rust. From [the language guide] I can gather, that it can fill up default values in a very generic way, but how can it know that when Rust prepared the messaged, that "cardCountsSeparateInactive" would default to true?

Reply:
Protobuf can make messages shorter by skipping keys when they are set to zero/empty, as that's the default value in protobuf. If we look at the message in Python, note how it only shows the non-zero key, but if we explicitly request another key, we do get a value:

```text
>>> from anki.backend_pb2 import GraphPreferences as G
... g = G()
... g.ParseFromString(mw.col.backend.get_graph_preferences())
... print(g)
... print("**")
... print(g.calendar_first_day_of_week)
card_counts_separate_inactive: true

**
0
```

the defaults arg in toObject() is necessary to create an object that has all keys set, even if they are zero, so we can iterate over them. So there are two separate "defaults" here - the logical value we think each key should start with, which is controlled by config.rs - and the zero/empty value that protobuf uses on the wire.